### DISCUSSION TITLE
Status and next steps

### DISCUSSION THREAD

**Discussion by micheltucker** [threadsjs/threads.js#43] [General]
@elijah-wright, as you started the repo. What are your next steps? Discover more endpoints? Get a /fetchall API going? Would love to collaborate on a list of possible things to do.

Top of mind would be for me:

- /fetch all endpoint + pagination
- handle 2fa (not sure what the latest is)

From: https://github.com/threadsjs/threads.js/issues/42

**Comment by stevenlafl** [comment]
I finally got my local setup working so I'll be assisting where I can. I got 2 more functions in PR. But I agree about pagination, I implemented it manually and it seems to work based on what @elijah-wright said, but I will confirm by testing it myself tomorrow and make another PR

2A accounts should work as well with no changes, according to testing and @elijah-wright's comments.

I'm thinking a series of unit tests might also be another good step.

**Reply by micheltucker** [reply]
And how reliable do we think this setup will be. That is, what are the chances of this getting shut down by meta?

**Reply by stevenlafl** [reply]
Short term, I can't see it getting shut down. Limits and paid access were things Twitter were criticized for recently, which Meta capitalized on. Meta needs the engagement to drive adoption. I saw an article in Forbes estimating $8 billion in revenue by 2025, which is $3 billion higher than Twitter was. Meta can more than afford to look at the content posted rather than the fact that it's potentially automated. These are also mobile APIs already so they behave similarly with a few differences. And the interoperability with Mastodon would make shutdowns or limits quickly get ridiculous.

**Reply by micheltucker** [reply]
Makes sense and agreed. Let's hope 🤞🏽
